src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -8028,6 +7990,200 @@ interface I19<T19_1, T19_2> where T19_1 : I19<T19_1, T19_2>, T19_2
                 );
         }
 
+        [Theory]
+        [CombinatorialData]
+        public void OperatorSignature_09([CombinatorialValues("==", "!=")] string op, bool isVirtual)
+        {
+            var (modifier, body) = GetModifierAndBody(isVirtual);
+
+            var source1 =
+@"
+interface I1<T1> where T1 : I1<T1>
+{
+    static bool operator " + op + @"(T1 x, bool y) => throw null;
+}
+
+interface I2<T2> where T2 : struct, I2<T2>
+{
+    static bool operator " + op + @"(T2? x, bool y) => throw null;
+}
+
+interface I3<T3> where T3 : I3<T3>
+{
+    static " + modifier + @" bool operator " + op + @"(T3 x, bool y)" + body + @"
+}
+
+interface I4<T4> where T4 : struct, I4<T4>
+{
+    static " + modifier + @" bool operator " + op + @"(T4? x, bool y)" + body + @"
+}
+
+class C5<T5> where T5 : C5<T5>.I6
+{
+    public interface I6
+    {
+        static " + modifier + @" bool operator " + op + @"(T5 x, bool y)" + body + @"
+    }
+}
+
+interface I7<T71, T72> where T72 : I7<T71, T72> where T71 : T72
+{
+    static " + modifier + @" bool operator " + op + @"(T71 x, bool y)" + body + @"
+}
+
+interface I8<T8> where T8 : I9<T8>
+{
+    static " + modifier + @" bool operator " + op + @"(T8 x, bool y)" + body + @"
+}
+
+interface I9<T9> : I8<T9> where T9 : I9<T9> {}
+
+interface I10<T10> where T10 : C11<T10>
+{
+    static " + modifier + @" bool operator " + op + @"(T10 x, bool y)" + body + @"
+}
+
+class C11<T11> : I10<T11> where T11 : C11<T11> {}
+
+interface I12
+{
+    static " + modifier + @" bool operator " + op + @"(int x, bool y)" + body + @"
+}
+
+interface I13
+{
+    static " + modifier + @" bool operator " + op + @"(I13 x, bool y)" + body + @"
+}
+";
+
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: _supportingFramework);
+            compilation1.GetDiagnostics().Where(d => d.Code is not ((int)ErrorCode.ERR_OperatorNeedsMatch or (int)ErrorCode.ERR_InterfacesCantContainConversionOrEqualityOperators or (int)ErrorCode.ERR_UnimplementedInterfaceMember)).Verify(
+                // (4,26): error CS0563: One of the parameters of a binary operator must be the containing type
+                //     static bool operator ==(T1 x, bool y) => throw null;
+                Diagnostic(ErrorCode.ERR_BadBinaryOperatorSignature, op).WithLocation(4, 26),
+                // (9,26): error CS0563: One of the parameters of a binary operator must be the containing type
+                //     static bool operator ==(T2? x, bool y) => throw null;
+                Diagnostic(ErrorCode.ERR_BadBinaryOperatorSignature, op).WithLocation(9, 26),
+                // (26,39): error CS9045: One of the parameters of an equality, or inequality operator declared in interfaces must be a type parameter on the enclosing type constrained to the enclosing type
```

Comment:
Nit: the error code here doesn't line up with the actual number. Consider doing a find/replace in this file to clean it up (I initially did a search for 9046 was very confused when I didn't see any tests using it).